lib/routes/utils/oauth-mint.js

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const jwtool = require('fxa-jwtool')
+const Pool = require('../../pool')
+
+// From the Firefox source code:
+// Having this parameter be short has limited security value and can cause
+// spurious authentication values if the client's clock is skewed and
+// we fail to adjust. See Bug 983256.
+const ASSERTION_LIFETIME = 1000 * 3600 * 24 * 365 * 25 // 25 years in ms.
+const CERT_LIFETIME = 1000 * 3600 * 5 // 5 hours in ms.
+
+const CLIENT_ID = '5882386c6d801776' // Desktop CLIENT_ID, is this ok?
+
+module.exports = (config, signer) => {
+  const secretKey = jwtool.JWK.fromFile(config.secretKeyFile)
+  const publicKey = jwtool.JWK.fromFile(config.publicKeyFile)
+  const pool = new Pool(config.oauth.url, { timeout: 1000 })
+  return {
+    getOAuthToken(sessionToken, scope) {
+      return this.generateOAuthAssertion(sessionToken)
+        .then(assertion => {
+          return pool.post('/v1/authorization', {
+            assertion,
+            client_id: CLIENT_ID,
+            response_type: 'token',
+            scope
+          })
+        })
+    },
+    generateOAuthAssertion(sessionToken) {
+      return signer.sign({
+        publicKey: publicKey.toJSON(),
+        email: `${sessionToken.uid.toString('hex')}@${config.domain}`,
+        domain: config.domain,
+        duration: CERT_LIFETIME,
+        generation: sessionToken.verifierSetAt,
+        lastAuthAt: sessionToken.lastAuthAt(),
+        verifiedEmail: sessionToken.email,
```

Comment:
Please also include the new `authenticationMethods` and `authenticatorAssuranceLevel` fields introduced in https://github.com/mozilla/fxa-auth-server/pull/2346; in fact I wonder if it's worth pulling out the code here and in the `/certificate/sign` route to a common helper that just takes a `sessionToken` and returns the signed certificate.

Reply:
Which values should these two fields take? From reading the PR I'm a bit unsure what we want here since it is the server that is requesting that token.

Reply:
The server is requesting it on behalf of a particular user with a particular sessionToken, so I think it makes sense to use the values  directly from the sessionToken, just like you're doing with `lastAuthAt` above.